src/Framework/NativeMethods.cs

```diff
@@ -895,9 +822,11 @@ private static SystemInformationData SystemInformation
     [SupportedOSPlatform("windows")]
     internal static extern void GetNativeSystemInfo(ref SYSTEM_INFO lpSystemInfo);
 
-    [DllImport("kernel32.dll", SetLastError = true)]
+    const ushort ALL_PROCESSOR_GROUPS = 0xFFFF;
+
+    [DllImport("kernel32.dll")]
```

Comment:
No `SetLastError = true`?

Reply:
No because when I disassembled the function it only sets 2 error codes for either passing a value > maximum groups (doesn't apply in our case because we're using ALL_PROCESSOR_GROUPS) and out-of-memory (not possible with paging).

I've never seen the function fail in production but just to be sure I partially reverted the changes so when it returns 0 for failure we fallback to Environment.ProcessorCount 👍